shapely/constructive.py

```diff
@@ -982,8 +987,17 @@ def voronoi_polygons(
     <MULTILINESTRING ((3 4, 3 0))>
     >>> voronoi_polygons(Point(2, 2))
     <GEOMETRYCOLLECTION EMPTY>
+    >>> voronoi_polygons(points, ordered=True)
+    <GEOMETRYCOLLECTION (POLYGON ((0 0, 0 4, 3 4, 3 0, 0 0)), POLYGON ((6 4, 6 0...>
     """
-    return lib.voronoi_polygons(geometry, tolerance, extend_to, only_edges, **kwargs)
+    if ordered and lib.geos_version < (3, 12, 0):
```

Comment:
Given that strictly speaking you are allowing `ordered` to be an array like, this `if ordered` will fail with the "ambiguous" error. 

But I am also fine with just saying that `ordered` needs to be a scalar.